guide/self_hosted_registry.md

```diff
@@ -0,0 +1,159 @@
+# Using a self-hosted registry with OpenFaaS
+
+If you're using OpenFaaS on single host, then you don't need to push your images to a registry. They will just be used from the local Docker library.
+
+If you are using a remote server or a multi-node cluster then you can push your function's image to a registry or the Docker Hub.
+
+This describes how to use OpenFaaS in a swarm with your own local registry for hosting function images.
+
+## Set up a swarm
+
```

Comment:
Is it worth linking to the deployment guide rather than repeating the setup?

I can see us making a change to this then having to update 10+/- documents and bogs.

Reply:
Its tricky determining the balance in these docs of which parts should be copies and which parts should be references. Perhaps I can link to the doc and then also copy the content, sort of like I did in the kong k8s guide.
https://github.com/jonocodes/faas/blob/bdfdd5e9ffc5b255dee4bd3828a47b0c1fa955bb/guide/kong_integration_k8s.md#bring-up-kubernetes